### PR TITLE
OptionsPicker: Fix value undefined

### LINKED ISSUE #32339: When write text directly to a variable input field, then press enter not work and throw a error
**What happened**:

When write text directly to a variable input field, if the text not in the defalult value list, then press enter will throw an Error.

https://user-images.githubusercontent.com/41524013/112567483-dc869200-8e1b-11eb-9505-78e7cd302202.mov

**What you expected to happen**:

After press enter, the input text as the value of the variable. 

**How to reproduce it (as minimally and precisely as possible)**:

It is easy to reproduce it.

**Environment**:
- Grafana version: 7.3.5
- Data source type & version: opentsdb
- OS Grafana is installed on: mac os
- User OS & Browser: chrome
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by ldhnam**
**What this PR does / why we need it**:

This PR fixes value undefined.

**Which issue(s) this PR fixes**:

Fixes #32339

### CODE REVIEW

File: public/app/features/variables/pickers/OptionsPicker/reducer.ts [hunk 1]
```diff
@@ -150,7 +150,7 @@ const optionsPickerSlice = createSlice({
       const { multi, selectedValues } = state;
       const selected = !selectedValues.find((o) => o.value === option.value);
 
-      if (option.value === ALL_VARIABLE_VALUE || !multi || clearOthers) {
+      if ((option && option.value === ALL_VARIABLE_VALUE) || !multi || clearOthers) {
```
Comment (hugohaggmark) [inline public/app/features/variables/pickers/OptionsPicker/reducer.ts:153]: Maybe change to
```suggestion
      if ((option?.value === ALL_VARIABLE_VALUE) || !multi || clearOthers) {
```

**Review by hugohaggmark — COMMENTED** [review]
Looks good but you need to add tests as well. Thank you!

File: public/app/features/variables/pickers/OptionsPicker/reducer.ts (hunk 1 above)
Comment (ldhnam) [reply public/app/features/variables/pickers/OptionsPicker/reducer.ts:153]: Thanks for suggesting, i will change it

**Comment by dessen-xu** [comment]
This way to fix not work, still throw an error. 
image

Row 147 throw an error, he fix after.

**Comment by dessen-xu** [comment]
I had test.

**Comment by ldhnam** [comment]
Thanks all for reviewing my PR, i will fix this bug and add tests in another PR.

**Pull request closed without merging**